### PR TITLE
Extensibility by pub/sub and recovery password pages

### PR DESCRIPTION
**Pull request by pahaz**
Features:
 - password recovery page
 - change password page
 - auth forms autofill by url qs
 - extensibility schema by pub/sub pattern
 - send mail example

### CODE REVIEW

File: apps/_example05app/pages/auth/forgot.jsx [hunk 1]
```diff
@@ -1,23 +1,129 @@
...
+    wrapperCol: { offset: 8, span: 16 },
+}
+
+const ForgotForm = () => {
+    const [form] = Form.useForm()
+    const intl = useIntl()
+    const [isLoading, setIsLoading] = useState(false)
+    const [isSuccessMessage, setIsSuccessMessage] = useState(false)
+    const [startPasswordRecovery, ctx] = useMutation(START_PASSWORD_RECOVERY_MUTATION)
+    let initialValues = (typeof window !== 'undefined' && window.location && window.location.href.includes('?') ? qs.parse(window.location.href.split('?', 2)[1]) : {})
```
Comment (toplenboren) [inline apps/_example05app/pages/auth/forgot.jsx:37]: We have duplicated code here, maybe we can create a function and move it away to something like `/utils`?

File: packages/@core.keystone/schema.js [hunk 2]
```diff
@@ -50,14 +52,25 @@ class GQLListSchema {
...
+        const mergedSchema = { ...this.schema, ...schema }
+        Object.keys(schema).forEach((key) => {
+            if (isNotNullObject(schema[key]) && isNotNullObject(this.schema[key])) {
+                mergedSchema[key] = { ...this.schema[key], ...schema[key] }
+            }
         })
+        return new GQLListSchema(this.name, mergedSchema)
+    }
+
+    on (eventName, listener) {
```
Comment (Dimitreee) [inline packages/@core.keystone/schema.js:64]: Ты может уже рассказывал об этом, но я забыл. Тут мы подписываемся на изменение модели?

**Comment by Dimitreee** [comment]
Тесты бы починить еще

File: packages/@core.keystone/schema.js (hunk 2 above)
Comment (pahaz) [reply packages/@core.keystone/schema.js:64]: Yes, it's a new feature. You can use the pub/sub pattern to extend some base schema logic.

Comment (pahaz) [reply packages/@core.keystone/schema.js:64]: https://github.com/8iq/nodejs-hackathon-boilerplate-starter-kit/pull/1/files#diff-d23bf4bff74920c201390bc49d621f9bR14-R16 -- an override example of how-to subscribe

https://github.com/8iq/nodejs-hackathon-boilerplate-starter-kit/pull/1/files#diff-ac57bd97bcce23fd9fb01734c7e08177R229-R233 -- an example of how to emit event `@core`

File: apps/_example05app/pages/auth/forgot.jsx (hunk 1 above)
Comment (pahaz) [reply apps/_example05app/pages/auth/forgot.jsx]: Ok: extracted with tests
https://github.com/8iq/nodejs-hackathon-boilerplate-starter-kit/pull/1/commits/4c2243e782092e57d7658584621f20dfdb0d11a7

**Pull request merged**